pkg/ccl/logictestccl/tests/local-read-committed/local-read-committed_test: TestReadCommittedLogic_upsert_non_metamorphic failed

pkg/ccl/logictestccl/tests/local-read-committed/local-read-committed_test.TestReadCommittedLogic_upsert_non_metamorphic [failed](https://tanzanite.cluster.engflow.com/invocations/default/1a2a67c9-bb6b-4a08-b7d2-5130afd925be?testReportRun=1&testReportShard=48&testReportAttempt=1#targets-Ly9wa2cvY2NsL2xvZ2ljdGVzdGNjbC90ZXN0cy9sb2NhbC1yZWFkLWNvbW1pdHRlZDpsb2NhbC1yZWFkLWNvbW1pdHRlZF90ZXN0) on master @ [39e43b85ec3b02bc760df10fce1c19d09419d6f2](https://github.com/cockroachdb/cockroach/commits/39e43b85ec3b02bc760df10fce1c19d09419d6f2):

Fatal error:

```
panic: test timed out after 1h16m40s
running tests:
	TestReadCommittedLogic_upsert_non_metamorphic (1h15m27s)

```
Stack: 

```
goroutine 395141 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x265
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x45
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestReadCommittedLogic_upsert_non_metamorphic
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestReadCommittedLogic_upsert_non_metamorphic2726162930
    test_log_scope.go:76: use -show-logs to present logs inline
    test_server_shim.go:139: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
[13:11:08] setting distsql_workmem='90610B';
[13:11:11] --- progress: /var/lib/engflow/worker/work/2/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/local-read-committed/local-read-committed_test_/local-read-committed_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/upsert_non_metamorphic: 4 statements
[13:13:09] --- progress: /var/lib/engflow/worker/work/2/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/local-read-committed/local-read-committed_test_/local-read-committed_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/upsert_non_metamorphic: 9 statements
[13:18:54] --- progress: /var/lib/engflow/worker/work/2/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/local-read-committed/local-read-committed_test_/local-read-committed_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/upsert_non_metamorphic: 10 statements
[13:18:57] --- progress: /var/lib/engflow/worker/work/2/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/local-read-committed/local-read-committed_test_/local-read-committed_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/upsert_non_metamorphic: 12 statements
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=48</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #134792 pkg/ccl/logictestccl/tests/local-read-committed/local-read-committed_test: TestReadCommittedLogic_upsert_non_metamorphic failed [C-test-failure O-robot T-sql-queries branch-release-24.2 release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReadCommittedLogic_upsert_non_metamorphic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-44386

pkg/ccl/logictestccl/tests/local-read-committed/local-read-committed_test.TestReadCommittedLogic_upsert_non_metamorphic [failed](https://tanzanite.cluster.engflow.com/invocations/default/bc997cd8-e2b6-4f8d-aba9-c9b64088a348?testReportRun=3&testReportShard=48&testReportAttempt=1#targets-Ly9wa2cvY2NsL2xvZ2ljdGVzdGNjbC90ZXN0cy9sb2NhbC1yZWFkLWNvbW1pdHRlZDpsb2NhbC1yZWFkLWNvbW1pdHRlZF90ZXN0) on master @ [562232945d7bd199d1d07f067708072b4420f90a](https://github.com/cockroachdb/cockroach/commits/562232945d7bd199d1d07f067708072b4420f90a):

Fatal error:

```
panic: test timed out after 1h16m40s
running tests:
	TestReadCommittedLogic_upsert_non_metamorphic (1h15m35s)

```
Stack: 

```
goroutine 394519 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x265
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x45
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestReadCommittedLogic_upsert_non_metamorphic
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestReadCommittedLogic_upsert_non_metamorphic28127569
    test_log_scope.go:76: use -show-logs to present logs inline
    test_server_shim.go:139: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
[12:07:42] setting distsql_workmem='32777B';
[12:07:45] --- progress: /var/lib/engflow/worker/work/2/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/local-read-committed/local-read-committed_test_/local-read-committed_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/upsert_non_metamorphic: 4 statements
[12:09:38] --- progress: /var/lib/engflow/worker/work/2/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/local-read-committed/local-read-committed_test_/local-read-committed_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/upsert_non_metamorphic: 9 statements
[12:16:49] --- progress: /var/lib/engflow/worker/work/2/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/local-read-committed/local-read-committed_test_/local-read-committed_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/upsert_non_metamorphic: 10 statements
[12:16:57] --- progress: /var/lib/engflow/worker/work/2/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/local-read-committed/local-read-committed_test_/local-read-committed_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/upsert_non_metamorphic: 12 statements
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=3</code>
 - <code>shard=48</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReadCommittedLogic_upsert_non_metamorphic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/sql/logictest/testdata/logic_test/upsert_non_metamorphic
+++ b/pkg/sql/logictest/testdata/logic_test/upsert_non_metamorphic
@@ -4,7 +4,11 @@
 # limiting (#102472).
 statement ok
 CREATE TABLE src (s STRING);
+
+statement ok
 CREATE TABLE dest (s STRING);
+
+statement ok
 INSERT INTO src SELECT repeat('a', 100000) FROM generate_series(1, 60)
 
 user host-cluster-root
@@ -36,6 +40,8 @@ user root
 
 statement ok
 DROP TABLE src;
+
+statement ok
 DROP TABLE dest
 
 # Regression test for UPSERT batching logic (#51391).
@@ -48,7 +54,11 @@ user root
 
 statement ok
 CREATE TABLE src (s STRING);
+
+statement ok
 CREATE TABLE dest (s STRING);
+
+statement ok
 INSERT INTO src
 SELECT
 	'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa'
@@ -69,11 +79,15 @@ user root
 
 statement ok
 DROP TABLE src;
+
+statement ok
 DROP TABLE dest
 
 # Regression test for finishing UPSERT too early (#54456).
 statement ok
 CREATE TABLE t54456 (c INT PRIMARY KEY);
+
+statement ok
 UPSERT INTO t54456 SELECT i FROM generate_series(1, 25000) AS i
 
 query I

